torch/_functorch/vmap.py

```diff
@@ -130,21 +130,30 @@ def _create_batched_inputs(
                       for in_dim, arg in zip(flat_in_dims, flat_args)]
     return tree_unflatten(batched_inputs, args_spec)
 
-# Undos the batching (and any batch dimensions) associated with the `vmap_level`.
+
+def _maybe_remove_batch_dim(name, batched_output, vmap_level, batch_size, out_dim):
+    if out_dim is None:
+        if isinstance(batched_output, torch.Tensor):
+            raise ValueError(
+                f'vmap({name}, ...): `{name}` can not return a '
+                f'Tensor when out_dim is None'
```

Comment:
I think the following snippet won't work but it should.

```python
def fn(dummy):
    return torch.randn(3)

vmap(fn, out_dims=(None,))(dummy)
```

ref: https://github.com/pytorch/functorch/issues/439#issuecomment-1029291469

Reply:
Also, considering the above case, I think the message should say `can not return a BatchedTensor when out_dim is None`

Reply:
fixed and added a test case